server/src/main/java/dev/slimevr/bridge/ProtobufBridge.java

```diff
@@ -125,6 +130,132 @@ protected void writeTrackerUpdate(Tracker localTracker) {
+		TrackerRole role = localTracker.getTrackerPosition().getTrackerRole();
+
+		Tracker primaryTracker = null;
+		Tracker secondaryTracker = null;
+		Tracker tertiaryTracker = null;
+
+		// Given what the role is of localTracker, the tracker positions that
+		// make up that role are set to primaryTracker, secondaryTracker, and
+		// tertiaryTracker respectively.
+		primaryTracker = TrackerUtils
+			.getNonInternalTrackerForBodyPosition(
+				allTrackers,
+				TrackerPosition.getByTrackerRole(role)
+			);
+		switch (role) {
+			case WAIST:
+				secondaryTracker = TrackerUtils
+					.getNonInternalTrackerForBodyPosition(allTrackers, TrackerPosition.WAIST);
+				break;
+			case LEFT_FOOT:
+				secondaryTracker = TrackerUtils
+					.getNonInternalTrackerForBodyPosition(
+						allTrackers,
+						TrackerPosition.LEFT_LOWER_LEG
+					);
+				break;
+			case RIGHT_FOOT:
+				secondaryTracker = TrackerUtils
+					.getNonInternalTrackerForBodyPosition(
+						allTrackers,
+						TrackerPosition.RIGHT_LOWER_LEG
+					);
+				break;
+			case LEFT_ELBOW:
+				secondaryTracker = TrackerUtils
+					.getNonInternalTrackerForBodyPosition(
+						allTrackers,
+						TrackerPosition.LEFT_LOWER_ARM
+					);
+				tertiaryTracker = TrackerUtils
+					.getNonInternalTrackerForBodyPosition(
+						allTrackers,
+						TrackerPosition.LEFT_SHOULDER
+					);
+				break;
+			case RIGHT_ELBOW:
+				secondaryTracker = TrackerUtils
+					.getNonInternalTrackerForBodyPosition(
+						allTrackers,
+						TrackerPosition.RIGHT_LOWER_ARM
+					);
+				tertiaryTracker = TrackerUtils
+					.getNonInternalTrackerForBodyPosition(
+						allTrackers,
+						TrackerPosition.RIGHT_SHOULDER
+					);
+				break;
+		}
+
+		// If the battery level of the tracker is lower than lowestLevel, then
+		// the battery level of the tracker position becomes lowestLevel.
+		// Tracker voltage is set if the tracker position has a battery level
+		// lower than lowest level and has a battery voltage (owoTrack devices
+		// do not).
+		if ((primaryTracker != null) && (primaryTracker.getBatteryLevel() != null)) {
+			lowestLevel = primaryTracker.getBatteryLevel();
+
+			if (primaryTracker.getBatteryVoltage() != null) {
+				trackerVoltage = primaryTracker.getBatteryVoltage();
+			}
+		}
+		if (
+			(secondaryTracker != null)
+				&& (secondaryTracker.getBatteryLevel() != null)
+				&& (secondaryTracker.getBatteryLevel() < lowestLevel)
+		) {
+			lowestLevel = secondaryTracker.getBatteryLevel();
+
+			if (secondaryTracker.getBatteryVoltage() != null) {
+				trackerVoltage = secondaryTracker.getBatteryVoltage();
+			}
+		}
+		if (
+			(tertiaryTracker != null)
+				&& (tertiaryTracker.getBatteryLevel() != null)
+				&& (tertiaryTracker.getBatteryLevel() < lowestLevel)
+		) {
+			lowestLevel = tertiaryTracker.getBatteryLevel();
+
+			if (tertiaryTracker.getBatteryVoltage() != null) {
+				trackerVoltage = tertiaryTracker.getBatteryVoltage();
+			}
+		}
+
+		if (lowestLevel == 200) {
```

Comment:
```suggestion
		if (lowestLevel >= 200) {
```

Reply:
lowestLevel here is initialized to 200. Practically speaking, it realistically shouldn't go above 100% unless there's some error with battery percentage reporting. So, if after going through every check it's still "200%", the trackers in question don't report any battery level. There's probably a nicer way to go about doing this, like setting it as null I suppose.

Reply:
I understand the purpose, this reduces the number of checks required within the code. I just simply propose this, as *any* value over 200 is unreasonable and can be entirely assumed to be invalid.

Reply:
Ah, I see what you mean.